chiselFrontend/src/main/scala/chisel3/core/Module.scala

```diff
@@ -63,6 +65,19 @@ extends HasId {
   /** Legalized name of this module. */
   final val name = Builder.globalNamespace.name(desiredName)
 
+  /** FIRRTL Module name */
+  private[chisel3] var _modName = name
```

Comment:
The main reason this is problematic is that it's initialized to name, rather than being an Option[String] initialized to None.  With that change, I don't see a clear advantage or disadvantage to firrtlNameMap vs. the var.  (This is kind of like the old debate about the refMap.)